stl/inc/functional

```diff
@@ -2204,7 +2204,9 @@ _CONSTEXPR20 pair<_FwdItHaystack, _FwdItHaystack> _Search_pair_unchecked(
         }
 
         return {_Last1, _Last1};
-    } else if constexpr (_Is_fwd_iter_v<_FwdItHaystack> && _Is_fwd_iter_v<_FwdItPat>) {

```

Comment:
I am not a fan of those drive by changes. The old style makes it more apparent that everything in the branch is valid and everything else is not.

Reply:
I disagree strongly. `_Is_fwd_iter_v<_FwdItHaystack> && _Is_fwd_iter_v<_FwdItPat>` are requirements, and thus are different than the other checks - the other checks just allow us to make the algorithm more efficient, this is necessary for us to do anything at all.

Reply:
Either way it's a `static_assert`, so I feel less strongly about this than both of you :joy_cat: That said, I find @strega-nil-ms's rationale for why the checks are different to be persuasive.

This cleanup might have merited a separate PR (as we should generally avoid mixing cleanup work with feature work), but I'm okay with this given how small the change is.